components/org.wso2.carbon.identity.multi.attribute.login/org.wso2.carbon.identity.multi.attribute.login.resolver.regex/src/main/java/org/wso2/carbon/identity/multi/attribute/login/resolver/regex/RegexResolver.java

```diff
@@ -138,6 +128,22 @@ private void resolveDistinctUsersForClaims(String loginAttribute, List<String> a
             }
         }
 
+        // Check the users from username by default if there is no regex for username claim.
+        Claim usernameClaim = claimManager.getClaim(UserCoreClaimConstants.USERNAME_CLAIM_URI);
+        if (allowedAttributes.contains(UserCoreClaimConstants.USERNAME_CLAIM_URI)
```

Comment:
Do we need to check for username claim in the allowed list i.e any future possibility to remove username

Reply:
With the API we can update the attribute list without the username claim . So it's better to check that the username is existing in the list.